plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -459,7 +459,12 @@ public void setColumnComment(ConnectorSession session, JdbcTableHandle handle, J
     @Override
     public void setColumnType(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle column, Type type)
     {
-        throw new TrinoException(NOT_SUPPORTED, "This connector does not support setting column types");
+        String sql = format(
+                "ALTER TABLE %s MODIFY COLUMN %s %s",
```

Comment:
ClikcHouse preserves other column definitions (e.g. default_expr, codec) when we omit them? I would recommend adding the test to `BaseClickHouseConnectorTest`.

Reply:
All the tables(nation, region, customer, orders) in the tpch database, seem to be of Engine family `Log` and `MODIFY TABLE` is not supported in Engine family Log in ClickHouse. 
Is it OK if I create another table with Engine family MergeTree